src/QsCompiler/DataStructures/SymbolTable.fs

```diff
@@ -170,14 +170,22 @@ type private PartialNamespace private
         | true, specs -> specs.Add spec // it is up to the namespace to verify the type specializations
         | false, _ -> CallableSpecializations.Add(cName, new List<_>([spec]))
 
+    /// Deletes the *explicitly* defined specialization at the specified location for the callable with the given name.
+    /// Does not delete specializations that have been inserted by the compiler, i.e. specializations whose location matches the callable declaration location.
+    /// Throws the standard key does not exist exception if no specialization for the callable with that name exists.
+    member internal this.RemoveCallableSpecialization (location : QsLocation) cName = 
+        match CallableDeclarations.TryGetValue cName with 
+        | true, (_, decl) when decl.Position = location.Offset && decl.Range = location.Range -> 0 
+        | _ -> CallableSpecializations.[cName].RemoveAll (fun (_, res) -> location.Offset = res.Position && location.Range = res.Range)
```

Comment:
The specializations (in the future) could potentially be specified in different files (if we are to support declaring specializations outside the declaration body). The symbol table is set up to handle that. Correspondingly, I could move the assertion to the namespace (opposed to here in the partial namespace, which is the part in a particular file) but I'd rather keep the logic in the namespace manager. I'll add a doc a line to the doc comment though.

Reply:
OK - just want to make sure I'm understanding this, but even if a callable had specializations defined in multiple files, a given location in a partial namespace or a given (source, location) pair in a namespace uniquely identifies one declared specialization, so the return value could still be at most 1?

Reply:
Probably. That depends a little bit on what we allow if we have type specializations, and how it's handled in the symbol table (whether one line of code can represent multiple different specializations - probably not, but I haven't sufficiently thought about it yet). Right now it is indeed at most one, since we do not delete auto-generated functor specializations.